db/migrate/20241205035402_change_default_for_badges_icon.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+class ChangeDefaultForBadgesIcon < ActiveRecord::Migration[7.2]
+  def change
+    change_column_default :badges, :icon, from: "fa-certificate", to: "certificate"
+
+    up_only { execute <<~SQL }
+        UPDATE badges
+        SET icon = 'certificate'
+        WHERE icon LIKE 'fa-certificate';
```

Comment:
Not quite sure I understand the question, but the removal of the `fa-` prefix has always been done as part of application logic before we pass it into the FA renderer since the upgrade from FA4 to FA5 back in 2018 ([this PR](https://github.com/discourse/discourse/pull/6673)). The deprecation message got recently introduced during the FA6 upgrade.

We do leave `fa` as part of the `class` HTML attribute in the fontawesome renderer (https://github.com/discourse/discourse/blob/32665cf9dd26a00b4f95cb43164aa7ccb17f2fa4/app/assets/javascripts/discourse-common/addon/lib/icon-library.js#L121), so that will continue to appear in sites.